pytorch_pfn_extras/training/extensions/timeline_trace.py

```diff
@@ -0,0 +1,106 @@
+    :meth:`pytorch_pfn_extras.profiler.TimeSummary.report` context manager.
+
+    Args:
+        trigger: Trigger that decides when to output the trace.
+            This is distinct from the trigger of this extension
+            itself. If it is a tuple in the form ``<int>, 'epoch'`` or
+            ``<int>, 'iteration'``, it is passed to :class:`IntervalTrigger`.
+        enable: Trigger that enables the tracing.
+            Note that since the extensions are executed at the end of an iteration
+            the tracer will be enabled from the iteration after
+            the trigger is fired. If it is a tuple in the form ``<int>, 'epoch'`` or
+            ``<int>, 'iteration'``, it is passed to :class:`IntervalTrigger`.
+        disable: Trigger that disables the tracing.
+            Note that since the extensions are executed at the end of an iteration
+            the tracer will be disabled from the iteration after
+            the trigger is fired. If it is a tuple in the form ``<int>, 'epoch'`` or
+            ``<int>, 'iteration'``, it is passed to :class:`IntervalTrigger`.
+        filename (str): Name of the log file under the output directory. It can
+            be a format string: the last result dictionary is passed for the
+            formatting. For example, users can use '{iteration}' to separate
+            the log files for different iterations. If the log name is None, it
+            does not output the log to any file.
+        writer (writer object, optional): must be callable.
+            object to dump the log to. If specified, it needs to have a correct
+            `savefun` defined. The writer can override the save location in
+            the :class:`pytorch_pfn_extras.training.ExtensionsManager` object
+        tracer (tracer object, optional):
+            object with the tracing logic.  If not specified, the default
+            tracer in the thread local storage with be used.
+    Args:
+        entries (list): list of str
+
+    Returns:
+        header (str): header string
+        templates (str): template string for print values.
+    """
+
+    def __init__(
+        self,
+        trigger: trigger_module.TriggerLike = (1, "epoch"),
+        filename: Optional[str] = None,
+        enable: Optional[trigger_module.TriggerLike] = None,
+        disable: Optional[trigger_module.TriggerLike] = None,
+        **kwargs: Any,
+    ):
+        self._tracer = kwargs.get("tracer", get_tracer())
+        self._enable = None
+        if enable is not None:
+            self._enable = trigger_module.get_trigger(enable)
+
+        self._disable = None
+        if disable is not None:
+            self._disable = trigger_module.get_trigger(disable)
+
+        self._trigger = trigger_module.get_trigger(trigger)
+
+        self._filename = "chrome_trace.json" if filename is None else filename
+        self._writer = kwargs.get("writer", None)
+
+    def _flush_trace(self, manager: ExtensionsManagerProtocol) -> None:
```

Comment:
It would be nice to be able to select a mode to synchronize the tracer and then flush it as a strict `flush_trace()`. I would appreciate a TODO comment as there are so many things to consider including performance issues.